.github/workflows/todoapp_pipeline.yaml

```diff
@@ -37,28 +37,10 @@ jobs:
       - name: Checkout files 
         uses: actions/checkout@v2
       - name: build docker
-        run: docker build --target production --tag ${{ secrets.DOCKER_HUB_ACCESS_USER }}/todo-app:prod-${{ github.sha }} .
+        run: docker build --target production --tag ${{ secrets.DOCKER_HUB_ACCESS_USER }}/todo-app:prod-latest .
       - name: Push to docker hub  
-        run: docker push ${{ secrets.DOCKER_HUB_ACCESS_USER }}/todo-app:prod-${{ github.sha }}
-      - name: Heroku login credentials
-        run: |
-          cat > ~/.netrc <<EOF
-            machine api.heroku.com
-              login ${{ secrets.HEROKU_EMAIL }}
-              password ${{ secrets.HEROKU_API_KEY }}
-            machine git.heroku.com
-              login ${{ secrets.HEROKU_EMAIL }}
-              password ${{ secrets.HEROKU_API_KEY }}
-          EOF
-      - name: Add Heroku remote
-        run: heroku git:remote --app ${{ secrets.HEROKU_APP_NAME }}
-      - name: Login into Heroku's registry
-        run: heroku container:login
-      - name: Tag for heroku
-        run: docker tag ${{ secrets.DOCKER_HUB_ACCESS_USER }}/todo-app:prod-${{ github.sha }} registry.heroku.com/${{ secrets.HEROKU_APP_NAME }}/web
-      - name: Push to heroku
-        run: docker push registry.heroku.com/${{ secrets.HEROKU_APP_NAME }}/web
-      - name: Release heroku
-        run: heroku container:release web
+        run: docker push ${{ secrets.DOCKER_HUB_ACCESS_USER }}/todo-app:prod-latest
+      - name: Azure restart and pull the latest version of docker container
+        run: curl -dH -X POST 'https://\$bearsxapp:$AZURE_PASSWORD@bearsxapp.scm.azurewebsites.net/api/registry/webhook' --fail
```

Comment:
This looks great, but the result that is being displayed in the pipeline suggests to me that this isn't actually working:
>  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100     1    0     0  100     1      0      1  0:00:01 --:--:--  0:00:01     1

If this was succeeding I'd expect to see a small body response stating where you can track the deployment. To me this suggests that GitHub Actions isn't quite interpreting your environment variables correctly. In particular, I _think_ it's not interpreting your password at all, possibly for some combination of the following reasons:

Is the environment variable actually set? Just because it's set as a secret doesn't mean it's there as an environment variable, so you may want something more like:
```yml
        run: curl -dH -X POST 'https://\$bearsxapp:${{ secrets.AZURE_PASSWORD }}@bearsxapp.scm.azurewebsites.net/api/registry/webhook' --fail
```

It could also be because your combination of escaping and single quotes isn't quite working; typically single quotes mean that the shell doesn't substitute environment variables (e.g. try `echo '\$PATH'` in a bash shell and you should see it literally prints `\$PATH`) so we might want to try switching to double quotes (`"`) _or_ it can be easier to swap for the entire URL in your secret, so we just do:
```
run: curl -dH -X POST '${{ secrets.WEBHOOK_URL }}' --fail
```
And that URL is set up with the password just included, and (if I remember rightly) does have the escaping `\`.

Reply:
Hi Jack,

thanks for the review. Sorry I am using person azure account. The app works perfectly in Azure. I had stopped when I asked for review, sorry. Before I make the changes, you recommended can you access the app here: http://bearsxapp.azurewebsites.net/.

Do I still need to make the changes?